### PR TITLE
Prepare 0.8.0 (versions)

### PR DESCRIPTION
**Pull request by jneira**
* This pr include version bumps for all components (hls itself, hls-plugin-api and ghcide)
* A hackage index update
* Also a fix to macos builds
* And a mention to lib versions and ghcide-0.6.0.2 changelog

Follow up of #1149

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 ## 0.8.0
 
 - This version adds support for ghc-8.10.3
+- `hls-plugin-api` is 0.6.0.0 and `ghcide`, 0.7.0.0
+  - so this include [changes from `ghcide-0.6.0.2`](https://github.com/haskell/haskell-language-server/blob/master/ghcide/CHANGELOG.md#0602-2020-12-26)
```
Comment (pepeiborra) [inline ChangeLog.md:7]:
```suggestion
- This version adds support for ghc-8.10.3
- `hls-plugin-api` has been bumped to 0.6.0.0 and `ghcide` has been bumped from 0.6.0.1 to 0.7.0.0.
```

**Comment by pepeiborra** [comment]
Thanks. Are you planning to update the ghcide changelog as well?

**Review by alanz — COMMENTED** [review]
LGTM

**Comment by jneira** [comment]
> 
> 
> Thanks. Are you planning to update the ghcide changelog as well?

We weren't update the changelog of each subcomponent until now and the changelog contains pr's affecting one o more of them at the same time. But maybe we could start to do it, at least for ghcide. 

should we keep those pr's in the main changelog and in ghcide one or only in the latter?

**Comment by jneira** [comment]
@pepeiborra i've added some entries in the ghcide changelog, keeping the main hls one instact. I've taken the list of [your comment in the previous pr](https://github.com/haskell/haskell-language-server/pull/1149#issuecomment-753545005), removing the existent ones.

**Comment by pepeiborra** [comment]
> > Thanks. Are you planning to update the ghcide changelog as well?
> 
> We weren't update the changelog of each subcomponent until now and the changelog contains pr's affecting one o more of them at the same time. But maybe we could start to do it, at least for ghcide.

I must be missing something. In #1149 you said that the ghcide changes should go in the ghcide changelog:

>@pepeiborra hi, thanks for the suggestion. However, some of those pr's are already included in the ghcide Changelog or in the actual list of merged pr's. The unique i miss from both lists is haskell/ghcide#952, that was merged before ghcide-0.6.0.2 but no listed in the changelog for some reason. It seems its place would be the ghcide Changelog.
https://github.com/haskell/haskell-language-server/pull/1149#issuecomment-753574081

That's why I am asking if you are planning to update the ghcide changelog before releasing it. 

> should we keep those pr's in the main changelog and in ghcide one or only in the latter?

I don't have an opinion. Probably a link is enough

**Review by pepeiborra — APPROVED** [review]

**Comment by jneira** [comment]
Mmm, i see, my intent was to add the missing info (#952) in the last version included in the ghcide changelog (0.6.0.2). With `It seems its place would be the ghcide Changelog.` i was referring only to that pr.

... and keep the rest of ghcide related pr's, merged after it inclusion in this repo, in the main hls changelog (as they were already included there at that point). But continuing the ghcide changelog seems sensible to me too.

**Review by Ailrun — APPROVED** [review]
LGTM

**Pull request merged**